keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,1047 @@
+-->
+
+We propose removing support for the in-tree gitRepo volume driver. We aren't
+proposing removing gitVOlumes from the Kubernetes API, meaning pods with gitRepo 
+volumes will be admitted by kube-apiserver but kubelets with the feature-gate 
+GitRepoVolumeDriver set to false will not run them and return an appropriate 
+error to the user.
+
+We acknowledge that this is highly unusual, but there are mitigating circumstances:
+* gitRepo volume types can be exploited to gain remote code execution as root on the nodes as shown in [CVE-2024-10220](https://nvd.nist.gov/vuln/detail/cve-2024-10220)
+* gitRepo has perfectly workable alternatives:
+  * Using [git-sync](https://github.com/kubernetes/git-sync)
+  * Using an initContainer. See https://kubernetes.io/docs/concepts/storage/volumes/#gitrepo for more details
+* gitRepo has been deprecated for a long time and is unmaintained
+* gitRepo has low usage (based on limited data)
+* there exists precedent for removing volume drivers which were unmaintained like [glusterfs](https://kubernetes.io/docs/concepts/storage/volumes/#glusterfs)
+
+Given this, we think kubernetes should EOL this feature with urgency.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Remove a vector for remote code execution through an ancient and unmaintained
+volume plugin.
+
+SIG Storage was actually very close to doing this and had even added a notice of
+removal in the release notes for v1.32. But due to concerns raised by SIG Architecture
+they rolled back the notice. SIG Storage provided [this rationale](https://github.com/kubernetes/kubernetes/issues/125983#issuecomment-2522201283) for considering removal.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Remove in-tree gitRepo volume support.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Remove gitRepo volumes from the Kubernetes API.
+
+## Proposal
+
+Kubernetes drops support for gitRepo volume types by removing the in-tree driver
+that actually mounts the volume in the pod. In this case we will still keep the
+gitRepo volume in the API. Pods with gitRepo volumes will be admitted by 
+kube-apiserver however, kubelet will fail to run them with an appropriate 
+error message.
+
+
+This removal is similar to removal of support for
+[glusterfs](https://kubernetes.io/docs/concepts/storage/volumes/#glusterfs)
+which was deprecated in v1.25 and removed in v1.26.
```

Comment:
Was there a KEP for this? Did we offer any migration like CSI migration?

Reply:
I don't think there was a KEP. I searched for the "glusterfs" under /keps/ and got only one hit but that KEP had nothing to do with removal.